thirdparty/github.go

```diff
@@ -32,54 +30,84 @@ const (
 	GithubAPIStatusGood  = "good"
 )
 
-type GithubUser struct {
-	Active       bool   `json:"active"`
-	DispName     string `json:"display-name"`
-	EmailAddress string `json:"email"`
-	FirstName    string `json:"first-name"`
-	LastName     string `json:"last-name"`
-	Name         string `json:"name"`
+func githubRetry(resp *github.Response, err error) (bool, error) {
+	if _, ok := err.(*github.RateLimitError); ok {
+		return false, err
+	}
+	if resp.StatusCode == http.StatusBadRequest {
+		return false, err
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		return false, err
+	}
+	if resp.StatusCode == http.StatusUnauthorized {
+		err = errors.Errorf("Calling github GET on %v failed: got 'unauthorized' response", resp.Request.URL.String())
+		grip.Error(err)
+		return false, err
+	}
+	if err != nil {
+		grip.Errorf("failed trying to call github GET on %s: %+v", resp.Request.URL.String(), err)
+		return true, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		err = errors.Errorf("Calling github GET %s got a bad response code: %v", resp.Request.URL.String(), resp.StatusCode)
+	}
+
+	rateMessage, _ := getGithubRateLimit(resp.Header)
+	grip.Debugf("Github API response: %s. %s", resp.Status, rateMessage)
+
+	return false, err
 }
 
 // GetGithubCommits returns a slice of GithubCommit objects from
 // the given commitsURL when provided a valid oauth token
-func GetGithubCommits(oauthToken, commitsURL string) (
-	githubCommits []github.RepositoryCommit, header http.Header, err error) {
-	resp, err := tryGithubGet(oauthToken, commitsURL)
-	if resp != nil {
-		defer resp.Body.Close()
-	}
-	if resp == nil {
-		errMsg := fmt.Sprintf("nil response from url '%v'", commitsURL)
-		grip.Error(errMsg)
-		return nil, nil, APIResponseError{errMsg}
-	}
+func GetGithubCommits(oauthToken, owner, repo, ref string, commitPage int) ([]*github.RepositoryCommit, int, error) {
+	httpClient, err := util.GetHttpClientForOauth2(oauthToken)
 	if err != nil {
-		errMsg := fmt.Sprintf("error querying '%v': %v", commitsURL, err)
-		grip.Error(errMsg)
-		return nil, nil, APIResponseError{errMsg}
+		return nil, 0, errors.Wrap(err, "can't fetch data from github")
 	}
+	defer util.PutHttpClientForOauth2(httpClient)
+
+	client := github.NewClient(httpClient)
+	var commits []*github.RepositoryCommit
+	var resp *github.Response
+	_, _ = util.Retry(
+		func() (bool, error) {
+			commits, resp, err = client.Repositories.ListCommits(context.TODO(), owner, repo, &github.CommitsListOptions{
+				SHA: ref,
+				ListOptions: github.ListOptions{
+					Page: commitPage,
+				},
+			})
+
+			return githubRetry(resp, err)
+		}, NumGithubRetries, GithubSleepTimeSecs*time.Second)
```

Comment:
Change `GithubSleepTimeSecs` to `time.Second` so you don't have to multiply. (Maybe original author didn't think a `const` could hold `time.Second`)?